packages/timelock-manager/contracts/TimelockManager.sol

```diff
@@ -260,53 +296,63 @@ contract TimelockManager is Ownable, ITimelockManager {
             );
         Timelock memory timelock = timelocks[indTimelock];
         require(
-            timelock.amount != 0,
+            timelock.releasedAmount < timelock.totalAmount,
             "Timelock is already withdrawn"
             );
         require(
             msg.sender == timelock.owner,
             "Only the owner of the timelock can withdraw from it"
             );
         // Do not check if the timelock has matured
-        delete timelocks[indTimelock].amount;
+        uint256 amountToTransfer = timelock.totalAmount.sub(timelock.releasedAmount);
+        timelocks[indTimelock].releasedAmount = timelock.totalAmount;
         emit WithdrawnToPool(
             indTimelock,
             api3PoolAddress,
             beneficiary
             );
-        api3Token.approve(address(api3Pool), timelock.amount);
-        // If (now > timelock.releaseTime), the beneficiary can immediately
+        api3Token.approve(address(api3Pool), amountToTransfer);
+        // If (now > timelock.releaseStart), the beneficiary can immediately
         // have their tokens vested at the pool with an additional transaction
         api3Pool.depositWithVesting(
```

Comment:
Big TODO here, we'll have to pass `now` as the pool `releaseStart` and `releaseEnd` as the pool `releaseEnd`. Let me know if I need to elaborate.